org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -1289,31 +1289,109 @@ class FileEvent {
 /**
  * Value-object describing what options formatting should use.
  */
-@JsonRpcData
-class FormattingOptions {
+class FormattingOptions extends LinkedHashMap<String, Object> {
```

Comment:
Maybe, but I don't like `Either<Boolean, Either<Number, String>>`. Should we add an `Either3` class so we can write `Either3<Boolean, Number, String>`?